services/catalog/tests/unit/with_dbs/test_api_routes_services__inaccessible.py

```diff
@@ -0,0 +1,369 @@
+            await conn.execute(users.delete().where(users.c.id == id_))
+
+
+async def test_inaccessible_services_for_primary_group(
+    mock_catalog_background_task: None,
+    director_mockup: MockRouter,
+    client: TestClient,
+    user_id: UserID,
+    products_names: list[str],
+    service_catalog_faker: Callable,
+    services_db_tables_injector: Callable,
+    sqlalchemy_async_engine: AsyncEngine,
+    users_data: list[(UserID, GroupID)],
+):
+    target_product = products_names[0]  # osparc
+    # create some fake services
+    NUM_SERVICES = 3
+    fake_services = [
+        service_catalog_faker(
+            "simcore/services/dynamic/jupyterlab",
+            f"1.0.{s}",
+            team_access=None,
+            everyone_access=None,
+            product=target_product,
+        )
+        for s in range(NUM_SERVICES)
+    ]
+    # injects fake data in db
+    await services_db_tables_injector(fake_services)
+
+    service_to_check = fake_services[0][0]  # --> service_meta_data table format
+
+    # Currently only the owner has access to the service
+    url = URL("/v0/services:inaccessible")
+    response = client.post(
+        f"{url}",
+        headers={"x-simcore-products-name": target_product},
+        json={
+            "services_to_check": [
+                {
+                    "key": service_to_check["key"],
+                    "version": service_to_check["version"],
+                },
+            ],
+            "with_gid": 1,
+        },
+    )
+    assert response.status_code == 200
+    data = parse_obj_as(list[UserInaccessibleService], response.json())
+    assert user_id not in {item.user_id for item in data}
+    assert len(data) == 4
+
+    # We share the service with other user primary group
+    insert_user_id, insert_user_primary_gid = users_data[0]
+    async with sqlalchemy_async_engine.begin() as conn:
+        await conn.execute(
+            services_access_rights.insert()
+            .values(
+                key=service_to_check["key"],
+                version=service_to_check["version"],
+                gid=insert_user_primary_gid,
+                execute_access=True,
+                write_access=True,
+            )
+            .returning(services_access_rights)
+        )
+
+    url = URL("/v0/services:inaccessible")
+    response = client.post(
+        f"{url}",
+        headers={"x-simcore-products-name": target_product},
+        json={
+            "services_to_check": [
+                {
+                    "key": service_to_check["key"],
+                    "version": service_to_check["version"],
+                },
+            ],
+            "with_gid": 1,
+        },
+    )
+    assert response.status_code == 200
+    data = parse_obj_as(list[UserInaccessibleService], response.json())
+    response_users = {item.user_id for item in data}
+    assert user_id not in response_users
+    assert insert_user_id not in response_users
+    assert len(data) == 3
```

Comment:
I guess all these check (and event the call to the endpoint ?) could be wrapped in a function for all the tests in this module

Reply:
We discussed there are small differences so probably is not needed in this self-contained test.